internal/logic/logic_test.go

```diff
@@ -51,73 +63,232 @@ func Test_transactionManagementServiceLogic_HealthCheck(t *testing.T) {
+	srv *httptest.Server
+	t   *testing.T
+	wg  *sync.WaitGroup
+}
+
+func testClient(c *TestServer) {
+	router := mux.NewRouter()
+	router.HandleFunc("/microbank/v1/account/update/transaction", func(w http.ResponseWriter, r *http.Request) {
+		defer c.wg.Done()
+		defer c.t.Log("Hit")
+		expectedReqBody, _ := json.Marshal(model.UpdateTransaction{
+			AccountNumber:   0,
+			Amount:          1000,
+			TransactionType: "debit",
+		})
+		reqBody, _ := ioutil.ReadAll(r.Body)
+		if string(reqBody) != string(expectedReqBody) {
+			c.t.Errorf("Expected request body '%s', got '%s'", expectedReqBody, reqBody)
+			c.t.Fail()
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+	}).Methods(http.MethodPut)
+	c.srv = httptest.NewServer(router)
+}
+
+func TestTransactionManagementServiceLogic_NewTransaction(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	tStruct := &TestServer{
+		t:  t,
+		wg: &sync.WaitGroup{},
+	}
+	tests := []struct {
+		name        string
+		credentials model.NewTransaction
+		setup       func() (datasource.DataSourceI, string)
+		want        func(*respModel.Response)
+	}{
+		{
+			name: "Success::transaction status != approved",
+			credentials: model.NewTransaction{
+				UserId: "123",
 			},
-			give: &model.PingRequest{
-				Data: "ping",
+			setup: func() (datasource.DataSourceI, string) {
+				mockDs := mock.NewMockDataSourceI(mockCtrl)
+				mockDs.EXPECT().Insert(gomock.Any()).Times(1).DoAndReturn(func(tr model.Transaction) error {
+					tr.TransactionId = ""
+					tr.CreatedAt = time.Time{}
+					diff := testutil.Diff(tr, model.Transaction{
+						UserId:        "123",
+						AccountNumber: 0,
+						Amount:        0,
+					})
+					if diff != "" {
+						t.Error(testutil.Callers(), diff)
+					}
+					return nil
+				})
+				return mockDs, ""
 			},
-			want: &respModel.Response{
-				Status:  http.StatusOK,
-				Message: "Pong",
-				Data: &model.DsResponse{
-					Data: "pong",
-				},
+			want: func(resp *respModel.Response) {
+				temp := respModel.Response{
+					Status:  http.StatusCreated,
+					Message: "SUCCESS",
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(resp, &temp) {
+					t.Errorf("Want: %v, Got: %v", temp, resp)
+				}
 			},
 		},
 		{
-			name: "Failure::datasource error",
-			setup: func() datasource.DataSource {
-				mockDs := mock.NewMockDataSource(mockCtrl)
-
-				mockDs.EXPECT().Ping(&model.PingDs{
-					Data: "ping",
-				}).Times(1).
-					Return(nil, errors.New("ds down"))
+			name: "Success::transaction status = approved",
+			credentials: model.NewTransaction{
+				UserId: "123",
+				Type:   "debit",
+				Status: "approved",
+				Amount: 1000,
+			},
+			setup: func() (datasource.DataSourceI, string) {
+				tStruct.wg.Add(1)
+				testClient(tStruct)
+				mockDs := mock.NewMockDataSourceI(mockCtrl)
+				mockDs.EXPECT().Insert(gomock.Any()).Times(1).Return(nil)
 
-				return mockDs
+				return mockDs, tStruct.srv.URL
+			},
+			want: func(resp *respModel.Response) {
```

Comment:
Where is the validation done if the mock account management server is hit?

Reply:
done

Reply:
> Where is the validation done if the mock account management server is hit?

done